src/components/modal/faqCard.tsx

```diff
@@ -1,21 +1,78 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
 import { closeIcon } from '../../assets';
 import TextInput from '../Input/TextInput';
 import TextArea from '../Input/TextArea';
 import IconWrap from '../ui/svgWrapper';
+import { useFormik } from 'formik';
+import { faqFormSchema } from './modal.schema';
+import {
+  useCreateFaqMutation,
+  useEditFaqMutation,
+  useGetOneFaqQuery,
+} from '../../redux/api/faq';
+import { toast } from 'react-toastify';
+import ButtonLoader from '../button/buttonLoader';
+import { useEffect } from 'react';
 
-const FaqCard = ({
-  close,
-  isEdit,
-}: {
-  close: () => void;
-  isEdit?: boolean;
-}) => {
-  const handleSubmit = (e: React.SyntheticEvent) => {
-    e.preventDefault();
-    close();
+interface IFaqData {
+  question: string;
+  answer: string;
+}
+
+const initialFaq = {
+  question: '',
+  answer: '',
+};
+
+const FaqCard = ({ close, id }: { close: () => void; id?: string }) => {
+  const [createFaq, { isLoading }] = useCreateFaqMutation();
+  const [editFaq, { isLoading: editing }] = useEditFaqMutation();
+  const { data: faq } = useGetOneFaqQuery(id as string, {
+    skip: id === undefined,
+  });
```

Comment:
Why not pass in faq from the table into this component rather than make another api call?